feat(node): Add postgresjs instrumentation

onurtemizkan opened the pull request:
Resolves: https://github.com/getsentry/sentry-javascript/issues/15621
Adds instrumentation for https://github.com/porsager/postgres 

Sampled event: (Emitted from the integration tests added here): [Link](https://sentry-sdks.sentry.io/insights/backend/summary/trace/72c94a37c9907cc2c7f4bef9c56b0196/?fov=0%2C32.09936037659645&node=span-b3505cfada7dea73&project=5429215&query=transaction.op%3Atransaction&referrer=insights-backend-overview&source=performance_transaction_summary&statsPeriod=5m&timestamp=1750718572&transaction=Test%20Transaction)

This implementation patches `connection` and `query` classes to create database transactions:

- From `connection`, we pick up the database `name`, `url` and `port` to use in the db query spans
- For each `query` instance, we create a `db` span
- This implementation does not create a separate span for each `cursor` used

Initially, I implemented a way to capture `db.operation` (as `command` is available when the query resolves) but it seems the ingestion extracts the operation anyway, so I removed it.

Also added sanitization/normalization for raw query, which we use as the span description, also seems to be normalized by the ingestion engine. We can remove it too if it's not worth having, as it creates a possibly-unnecessary performance overhead on the SDK side.

AbhiPrasad commented on packages/node/src/integrations/tracing/postgresjs.ts:
```diff
@@ -0,0 +1,239 @@
+        }[],
+      ) => {
+        span.setStatus({
+          code: SPAN_STATUS_ERROR,
```
should we attach the `rejectArgs` here too?

AbhiPrasad reviewed (commented):
Let's make sure we collect as much as possible as per https://opentelemetry.io/docs/specs/semconv/database/database-spans/

Similar to the pg instrumentation, let's implement the `requireParentSpan` option and set it as true.

We should also implement `requestHook` so we can attach span origin.

onurtemizkan commented:
@AbhiPrasad 

- Added `requireParentSpan` and `requestHook` for when it's needed. https://github.com/getsentry/sentry-javascript/pull/16665/commits/edcc0164f7e7bcdfbafb1ce483a183d58fa3c013
- Added Sentry origin inline instead of using the hook. https://github.com/getsentry/sentry-javascript/pull/16665/commits/edcc0164f7e7bcdfbafb1ce483a183d58fa3c013
- (re)Added a few more OTEL attributes https://github.com/getsentry/sentry-javascript/pull/16665/commits/9798064611e379ef3f98c0fc956249f50bfe9a94:
  - `ATTR_DB_OPERATION_NAME` - this one is set inside resolve as it's only available there.
  - `ATTR_DB_QUERY_TEXT` - This is the sanitized query we use as the span description.
  - `ATTR_DB_RESPONSE_STATUS_CODE` - Error code inside `reject`, set when it's available.
  - `ATTR_ERROR_TYPE`: Set inside `reject`

- Table / Relation name is not available anywhere without a need for parsing, so I could not set `ATTR_DB_COLLECTION_NAME` It's extracted by the ingestion engine as far as I can see.
- `ATTR_DB_QUERY_SUMMARY` needs parsing as well, so I left it out. But it's also parsed by the ingestion engine and set as `normalized_description`
